src/proto/batch/mod.rs

```diff
@@ -235,6 +239,34 @@ impl<'a, S: Read + Write> BatchHandle<'a, S> {
     }
 }
 
+// Not documented, but existing de fakto and also mentioned in the official client
+// https://github.com/contribsys/faktory/blob/main/client/batch.go#L17-L19
+/// State of a `callback` job of a [`Batch`].
+#[derive(Debug, Clone, Deserialize)]
+pub enum CallbackState {
```

Comment:
If we suspect this may grow to include something like a `FinishedNotOk` in the future, we could mark this with `#[non_exhaustive]` so that we _can_ add variants to it. The downside is that callers don't get exhaustive matching, which is also pretty sad.

Reply:
`FinishedNotOk` looks like a very likely candidate. Added `#[non_exhaustive]`